cmd/chain-indexing/projections.go

```diff
@@ -1,38 +1,37 @@
 package main
 
 import (
-	cosmosapp_infrastructure "github.com/crypto-com/chain-indexing/infrastructure/cosmosapp"
+	"strings"
 
-	"github.com/crypto-com/chain-indexing/appinterface/projection/account"
-	"github.com/crypto-com/chain-indexing/appinterface/projection/account_message"
-	"github.com/crypto-com/chain-indexing/appinterface/projection/block"
-	"github.com/crypto-com/chain-indexing/appinterface/projection/blockevent"
-	transaction "github.com/crypto-com/chain-indexing/appinterface/projection/transaction"
-	"github.com/crypto-com/chain-indexing/appinterface/projection/validator"
-	"github.com/crypto-com/chain-indexing/appinterface/projection/validatorstats"
 	"github.com/crypto-com/chain-indexing/appinterface/rdb"
 	projection_entity "github.com/crypto-com/chain-indexing/entity/projection"
+	cosmosapp_infrastructure "github.com/crypto-com/chain-indexing/infrastructure/cosmosapp"
 	applogger "github.com/crypto-com/chain-indexing/internal/logger"
+	"github.com/crypto-com/chain-indexing/projection"
 )
 
 func initProjections(
 	logger applogger.Logger,
 	rdbConn rdb.Conn,
 	config *Config,
 ) []projection_entity.Projection {
-	var consNodeAddressPrefix = config.Blockchain.ConNodeAddressPrefix
 	var cosmosAppClient = cosmosapp_infrastructure.NewHTTPClient(config.CosmosApp.HTTPRPCUL)
-	return []projection_entity.Projection{
-		block.NewBlock(logger, rdbConn),
-		transaction.NewTransaction(logger, rdbConn),
-		blockevent.NewBlockEvent(logger, rdbConn),
-		validator.NewValidator(
-			logger, rdbConn, consNodeAddressPrefix,
-		),
-		validatorstats.NewValidatorStats(logger, rdbConn),
-		account_message.NewAccountMessage(logger, rdbConn),
-		account.NewAccount(logger, rdbConn, cosmosAppClient, config.Blockchain.BaseDenom),
 
-		// register more projections here
+	projections := make([]projection_entity.Projection, 0, len(config.Projection.Enables))
+	initParams := projection.InitParams{
+		Logger:                logger,
+		RdbConn:               rdbConn,
+		CosmosAppClient:       cosmosAppClient,
+		ConsNodeAddressPrefix: config.Blockchain.ConNodeAddressPrefix,
```

Comment:
I think for a longer run, We shall instead of passing Address Prefixes we can pass `Network TYPE `value instead and let a central package manage address prefixes according to the network type.

Nothing major but will help.

Reply:
I tend to leave this option to be a config because in the long run we can serve not only Crypto.com Chain but can be used by another Cosmos SDK based chain